db/migrate/20180525171150_add_deleted_on_to_service_template.rb

```diff
@@ -0,0 +1,29 @@
+class AddDeletedOnToServiceTemplate < ActiveRecord::Migration[5.0]
+  class ServiceTemplate < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    include ReservedMigrationMixin
+    include MigrationStubHelper
+  end
+
+  def up
+    add_column :service_templates, :deleted_on, :datetime
+
+    say_with_time("Migrate data from reserved table to service_templates") do
+      ServiceTemplate.includes(:reserved_rec).each do |st|
+        st.reserved_hash_migrate(:deleted_on)
+      end
+    end
+  end
+
+  def down
+    say_with_time("Migrate data from service_templates to reserved table") do
+      ServiceTemplate.includes(:reserved_rec).each do |st|
+        st.reserved_hash_set(:deleted_on, st.deleted_on)
+        st.save!
```

Comment:
Is there no reversed migrate method?

Reply:
Nope just https://github.com/ManageIQ/manageiq-schema/blob/master/lib/reserved_migration_mixin.rb#L6-L10

Reply:
This is how all of our reserved table down migrations are today but I can write a reserved_hash_migrate down equivalent if you want.

`reserved_hash_unmigrate` or something?  It would just be `self.reserved_hash_set(k, v); self.save!`